docs/bmo-part-of-capm3.md

```diff
@@ -0,0 +1,281 @@
+Ironic are separated and as a consequence, it will result in breaking the chain
+of workflow, where, previously, BMO could use localhost to talk to ironic
+without any authentication. However, considering that Ironic runs in host
+network, BMO would be able to interact with it, from anywhere that is connected
+or routed to the provisioning network. Therefore, separating Ironic in its own
+pod should not impact BMO in this case.
+
+Furthermore, to support above provided statement, CAPM3  needs to embed BMO in a
+way that it is deployed and modified properly by `clusterctl` without having to
+embed ironic as part of the clusterctl. Thus, we created
+`cluster-api-provider-metal3/config/bmo/` folder to be able to deploy BMO in
+CAPM3. The following directory tree and kustomization file will give more
+explanation on what are we trying to achieve as an end-goal.
+
+```diff
+tree config/bmo/
+
+config/bmo/
+├── bmo_image_patch.yaml
+├── kustomization.yaml
+├── kustomizeconfig.yaml
+└── README.md
+
+```
+
+where `kustomization.yaml` file contains:
+
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: capm3-system
+resources:
+- github.com/metal3-io/baremetal-operator/deploy/operator/?ref=master
+- github.com/metal3-io/baremetal-operator/deploy/crds/?ref=master
+- github.com/metal3-io/baremetal-operator/deploy/rbac/?ref=master
+
+configMapGenerator:
+- behavior: create
+  literals:
+  - DEPLOY_KERNEL_URL=${DEPLOY_KERNEL_URL}
+  - DEPLOY_RAMDISK_URL=${DEPLOY_RAMDISK_URL}
+  - IRONIC_ENDPOINT=${IRONIC_URL}
+  - IRONIC_INSPECTOR_ENDPOINT=${IRONIC_INSPECTOR_URL}
+  name: ironic-bmo-configmap
+
+patchesStrategicMerge:
+- bmo_image_patch.yaml
+
+configurations:
+- kustomizeconfig.yaml
+
+```
+
+The kustomization file deploys `operator`, `rbac` and `crds` defined in
+`resources` section using cross repository referencing link and
+`bmo_image_patch.yaml` patches the default image URL through kustomization.
+`kustomizeconfig.yaml` teaches the kustomize where to look at when substituting
+variables. When the baremetal-operator is deployed through CAPM3, operator
+container creates the following environment variables through the
+configMapGenerator:
+
+### DEPLOY_KERNEL_URL
+
+This is the URL of the kernel to deploy. For example:
+
+`DEPLOY_KERNEL_URL="http://172.22.0.X:6180/images/ironic-python-agent.kernel"`
+
+### DEPLOY_RAMDISK_URL
+
+This is the URL of the ramdisk to deploy. For example:
+
+`DEPLOY_RAMDISK_URL="http://172.22.0.X:6180/images/ironic-python-agent.initramfs"`
+
+### IRONIC_URL
+
+This is the URL of the ironic endpoint. For example:
+
+`IRONIC_URL="http://172.22.0.X:6385/v1/"`
+
+### IRONIC_INSPECTOR_URL
+
+This is the URL of the ironic inspector endpoint.
+For example:
+
+`IRONIC_INSPECTOR_URL="http://172.22.0.X:5050/v1/"`
+
+where `X` can be `1` in case of launching a management cluster using `kind` or
+`2` in case of launching a management cluster using `minikube`.
```

Comment:
I wouldn't put a hardcoded IP here, considering that this could be anything when not running in dev env and that even in dev env it is configurable

Reply:
can we use `CLUSTER_URL_HOST` or `PROVISIONING_URL_HOST` instead of hardcoded IPs'? or any other suggestions @maelk ?

Reply:
since this is the proposal, I would just use a generic x.x.x.x or <ip-address>

Reply:
Updated